src/words/FrameLoop.java

```diff
@@ -87,8 +88,17 @@ private boolean executeSingleFrame() {
 		}
 
 		// Phase 3: Listener Evaluation
-		for (WordsEventListener eventListener : environment.getEventListeners()) {
-			eventListener.execute();
+		for (Iterator<WordsEventListener> iterator = environment.getEventListeners().iterator(); iterator.hasNext();) {
+			try {
+				WordsEventListener eventListener = iterator.next();
+				boolean delete = !eventListener.execute(environment);
+				if (delete) {
+					iterator.remove();
+				}
+			} catch (WordsRuntimeException e) {
+				System.err.println(e.toString());
+				System.out.println("> ");
```

Comment:
I don't really like how these System.out.println("> ") prompts are starting to profilerate all over our code.  I'm going to open an issue for someone to fix that.